### PR TITLE
Fix completion inside last part of long ident

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This fixes https://github.com/Microsoft/visualfsharp/issues/3709

Was:

image

Now:

image

### CODE REVIEW

**Comment by auduchinok** [comment]
Maybe we should move the QuickParse.fs to FCS? It is VS-agnostic and can be used in all tools.

**Comment by vasily-kirichenko** [comment]
@auduchinok done. Also I think parsing long idents should be hidden behind FCS API entirely. Stuff like `plids`, `residue` and `lastDotPos` should be removed, only line str and pos should be passed. It would simplify the API a lot, but I think it should be done in a separate PR.

**Comment by auduchinok** [comment]
@vasily-kirichenko thank you!
I absolutely agree about simplifying these APIs. I've been thinking about adding a type just like you did in this PR.

**Comment by vasily-kirichenko** [comment]
TBH, I didn't think much about that type, so it's not that elegant. The whole API needs a serious rethinking and simplifying based on real usage scenarios in Ionide, Rider and VS.

**Comment by vasily-kirichenko** [comment]
part3 failed with

```
09:20:07 D:\j\w\release_ci_pa---c6929ad7\src\fsharp\vs\service.fsi(149,162): error FS0039: The type 'PartialLongName' is not defined. [D:\j\w\release_ci_pa---c6929ad7\src\buildfromsource\FSharp.Compiler.Private\FSharp.Compiler.Private.fsproj]
```

It's built successfully from within VS on my machine. Any ideas?

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Windows_NT Release_ci_part2 Build please

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Windows_NT Release_ci_part2 Build please

**Comment by vasily-kirichenko** [comment]
It's ready

File: vsintegration/src/FSharp.Editor/Options/EditorOptions.fs [hunk 1]
```diff
@@ -13,7 +13,7 @@ module DefaultTuning =
     let UnusedDeclarationsAnalyzerInitialDelay = 0 (* 1000 *) (* milliseconds *)
     let UnusedOpensAnalyzerInitialDelay = 0 (* 2000 *) (* milliseconds *)
     let SimplifyNameInitialDelay = 2000 (* milliseconds *)
-    let SimplifyNameEachItemDelay = 5 (* milliseconds *)
+    let SimplifyNameEachItemDelay = 0 (* milliseconds *)
```
Comment (nosami) [inline vsintegration/src/FSharp.Editor/Options/EditorOptions.fs]: part of other PR?

Comment (vasily-kirichenko) [reply vsintegration/src/FSharp.Editor/Options/EditorOptions.fs]: Yes.

File: src/fsharp/vs/QuickParse.fsi [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      LastDotPos: int option }
+    static member Default : PartialLongName
+
+#if COMPILER_PUBLIC_API
+module QuickParse =
+#else
+module internal QuickParse =
+#endif
+    val MagicalAdjustmentConstant : int
+    val CorrectIdentifierToken : s: string -> tokenTag: int -> int
```
Comment (dsyme) [inline src/fsharp/vs/QuickParse.fsi]: Minor things
* Could you give better argument names please, e.g. ``tokenText`` not ``s``?  
* Also in the implementation.  
* Also a `///` comment on each public thing please
* Add a blank line between before each ``///`` comment section

File: src/fsharp/vs/service.fsi [hunk 3]
```diff
@@ -146,7 +146,7 @@ type internal FSharpCheckFileResults =
     ///    and assume that we're going to repeat the operation later on.
     /// </param>
     /// <param name="userOpName">An optional string used for tracing compiler operations associated with this request.</param>
-    member GetDeclarationListInfo : ParsedFileResultsOpt:FSharpParseFileResults option * line: int * colAtEndOfPartialName: int * lineText:string * qualifyingNames: string list * partialName: string * getAllSymbols: (unit -> AssemblySymbol list) * ?hasTextChangedSinceLastTypecheck: (obj * range -> bool) * ?userOpName: string -> Async<FSharpDeclarationListInfo>
+    member GetDeclarationListInfo : ParsedFileResultsOpt:FSharpParseFileResults option * line: int * colAtEndOfPartialName: int * lineText:string * partialName: PartialLongName * getAllSymbols: (unit -> AssemblySymbol list) * ?hasTextChangedSinceLastTypecheck: (obj * range -> bool) * ?userOpName: string -> Async<FSharpDeclarationListInfo>
```
Comment (dsyme) [inline src/fsharp/vs/service.fsi]: * Please fix up the ``param`` documentation so the parameter list match.   Also for `GetDeclarationListSymbols`

* Please add a sentence in the ``///`` comment text for ``partialName`` the saying that the ``QuickParse`` module can be used to get a ``partialName`` value

**Comment by vasily-kirichenko** [comment]
All done, waiting until it's green.

**Comment by vasily-kirichenko** [comment]
ready

**Pull request merged**